### PR TITLE
an alternative objective function can be part of a scenario and saved & much more

### PR DESCRIPTION
**Pull request by axelvonkamp**
scenario saves also the pinned reactions and a description
objective tab for editing/switching the objective functions
some glitches and bugs fixed

### CODE REVIEW

**Review by Paulocracy — CHANGES_REQUESTED** [review]
Thank you for the expanded scenario handling :-)
Generally, the objective override and description works great for me, I just encountered the following (partly subjective) issues:
* I didn't find a way to de-pin the scenario-edited reactions, is there a simple way to do it? Otherwise, maybe a toggle button (such as for the scenario coloring) might be useful UI-wise.
* The "Scenario" menu entry is now quite overwhelming :O Maybe we could discuss if there's a way to sort the entries, and which entries should be included there. E.g., loading and saving a scenario is not put in a subsequent order and "Show model bounds" could be moved somewhere else.

**Comment by axelvonkamp** [comment]
>     * I didn't find a way to de-pin the scenario-edited reactions, is there a simple way to do it? Otherwise, maybe a toggle button (such as for the scenario coloring) might be useful UI-wise.

You can unpin a reaction via the context menu associated with the reaction row in the reaction list. Or do you want to unpin multiple reactions?
 
>     * The "Scenario" menu entry is now quite overwhelming :O Maybe we could discuss if there's a way to sort the entries, and which entries should be included there. E.g., loading and saving a scenario is not put in a subsequent order and "Show model bounds" could be moved somewhere else.

Yes, it looks a bit overwhelming, if you want to rearrange it go ahead :-)

**Comment by Paulocracy** [comment]
Yes I meant multiple or just all pinned reactions at once.
And I'll add a proposal for a menu entry change.

**Comment by axelvonkamp** [comment]
> Yes I meant multiple or just all pinned reactions at once. And I'll add a proposal for a menu entry change.

Multiple selections in the reaction list are currently not supported so I made a new menu entry for unpinning all reactions.

File: cnapy/appdata.py [hunk 1]
```diff
@@ -140,7 +230,8 @@ def __init__(self):
 
         default_map = CnaMap("Map")
         self.maps = {"Map": default_map}
-        self.scen_values: Dict[str, Tuple[float, float]] = {}
+        self.scen_values: Scenario = Scenario()
+        # self.scen_values: Dict[str, Tuple[float, float]] = {}
```
Comment (sthiele) [inline cnapy/appdata.py:234]: You can get rid of this line.

Comment (Paulocracy) [reply cnapy/appdata.py]: Done by @axelvonkamp

**Review by Paulocracy — APPROVED** [review]
Thank you for all changes and improvements :-)

**Pull request merged**